Author: Buschmann <buschmann@huessenbergnetz.de>
Fixing an sql error when there is more than one device id for example because of more than one cover image in an album folder.

--- amarokFS-0.5.orig/amfs-xml.cpp	2007-03-19 13:35:38.000000000 +0100
+++ amarokFS-0.5/amfs-xml.cpp	2007-04-20 16:53:15.000000000 +0200
@@ -656,7 +656,11 @@ QImage AmfsXML::getCoverPath(QString art
 	QString coverPathQuery = "select path from images where artist='" + artist + "' and album='" + album + "'";
 	QStringList coverPath = getDCOPDataSQL(coverPathQuery);
 	
-	QString deviceMountPointQuery = "select lastmountpoint from devices where id=(select deviceid from images where artist='" + artist + "' and album='" + album + "')";
+	QString deviceMountPointIdQuery = "select deviceid from images where artist='" + artist + "' and album='" + album + "'";
+	QStringList deviceMountPointId = getDCOPDataSQL(deviceMountPointIdQuery);
+	QString deviceMountPointFirstId = deviceMountPointId.first();
+	
+	QString deviceMountPointQuery = "select lastmountpoint from devices where id='" + deviceMountPointFirstId + "'";
 	QStringList deviceMountPoint = getDCOPDataSQL(deviceMountPointQuery);
 	
 	//path - location relative within the media library
